locator/token_metadata.cc

```diff
@@ -85,6 +85,8 @@ class token_metadata_impl final {
     long _ring_version = 0;
     static thread_local long _static_ring_version;
 
+    token_metadata::version_t _version = 0;
+
```

Comment:
Should this be std::optional<> to distinguish between "we had a version 0" and "we did not read any version"?

Reply:
> In particular, in RPCs, if fencing_token.version == 0 we know the caller is not yet aware of versions/fencing

Wouldn't we receive `nullopt` in that case?